Feature: add yaml and markdown snippets

chrisluedtke opened the pull request:
Adds useful snippets for the context of writing dbt YAML and markdown files. To get this to work, I also had to modify my vscode settings ([as mentioned in this issue](https://github.com/Microsoft/vscode/issues/27095#issuecomment-339517188)):

```
  "editor.quickSuggestions": {
    "other": true,
    "comments": false,
    "strings": true
  }
```

### Caveats:
* Using the `jinja-yaml` language replaces the traditional `# ` comment style with `{# #}`

chrisluedtke committed:
add yaml snippets

chrisluedtke committed:
add newline

chrisluedtke committed:
standardize quoting

chrisluedtke committed:
update readmes

chrisluedtke committed:
add markdown snippet

chrisluedtke committed:
more standard doc snippet name

chrisluedtke committed:
remove space in docs snippet

bastienboutonnet commented:
Hi @chrisluedtke 
Thanks a lot for contributing to this! SOrry for the delay in replying I completely missed the notification. I'll have a look later today.

bastienboutonnet commented:
Hi again @chrisluedtke and thanks again for your contribution 🚀 !

I really like what I'm seeing here, and I had indeed not really yet felt the need to write snippets for yml.

One quick question, and pardon my ignorance. Can we actually use stuff like `{{ ref('something') }}` in yml? Like, does dbt do anything with it?

With regards to your caveat, well spotted. I don't see a huge problem, I guess if you were to use jinja in your yml you'd probably want to be able to have jinja-style comments? Although I guess the only issue is if people had some CI in place to validate their `.yml` files things would maybe complain. What do you think?

Actually thinking about it, the jinja-style comments are coming from the [`betterJinja`](https://marketplace.visualstudio.com/items?itemName=samuelcolvin.jinjahtml) extension. So I would think this concern has probably been raised and addressed there. Might be worth checking what the authors of this package have to say about it. I think for now, I think I'm not that concerned? In theory since you'll make a concious choice to use the jinja-yml language it means you'll probably worry about CI and other yml validation steps in your CI. But it could be unpleasent.

Curious to hear your thoughts.

Other than that, I think this is a pretty stellar PR and if we were both aligned on the caveat you pointed out I'd be more than happy to merge it and release the extension.

Thanks also for cleaning up and and updating the docs ✨ 

**EDIT**:
I looked at the betterJinja repo for issues
Something interesting about this going on: https://github.com/samuelcolvin/jinjahtml-vscode/issues/45#issuecomment-665383211

Maybe it's something I or we could look into contributing back to the betterJinja repo. For now I'm not sure and I wouldn't consider it a HUGE blocker. I'm actually just wondering. What makes the most sense? Should we want jinja-style comments in yml files at all?

**EDIT 1**:
Actually I tested the comment in a dbt file and it breaks compilation having this style so I guess now I am bothered about it :p It's gonna make some people suffer. I'll put up an issue against the betterJinja repo and see what the maintainer thinks (issue in the betterJinja repo: https://github.com/samuelcolvin/jinjahtml-vscode/issues/70). And see where to go from there.

Are you in any rush to have this PR merged? given what we know now? I don't want to block the work you did, but it's a bit risky indeed.

Happy to hear your side of the story.

bastienboutonnet commented:
Oh one more thing, regarding the VSCode settings you had to enable. Do you think there's a way to ship this modification as part of the extension? Would we even want to do that? Maybe good enough to add some documentation (although that's a bit clunky).

Since it's just a snippet's extension, it's not totally straighforward to build a settings UI but we could look into it

bastienboutonnet commented:
Actually, something interesting about this going on: https://github.com/samuelcolvin/jinjahtml-vscode/issues/45#issuecomment-665383211

Maybe it's something I or we could look into contributing back to the betterJinja repo. For now I'm not sure and I wouldn't consider it a HUGE blocker

chrisluedtke committed:
remove brackets from ref and source in YAML

chrisluedtke committed:
update readme

chrisluedtke commented:
> One quick question, and pardon my ignorance. Can we actually use stuff like {{ ref('something') }} in yml? Like, does dbt do anything with it?

@bastienboutonnet Yes, these are useful when adding relationship tests. I realized I mistakenly included brackets in the snippet initially, but I've removed them now.
```yml
- name: my_column
  description: "{{ doc('doc_for_my_column') }}"
  tests:
    - relationships:
        to: source('my_source', 'my_other_model')
        field: my_other_column
```

> Are you in any rush to have this PR merged? given what we know now? I don't want to block the work you did, but it's a bit risky indeed.

No rush here. I briefly looked into overriding the Jinja-style comment, but I don't think it's possible. We would probably need our own custom language or a modification upstream. Thank you for opening that issue!

I also just added some helpful notes on set-up, since it took me a while to figure it out.

chrisluedtke commented:
> Oh one more thing, regarding the VSCode settings you had to enable. Do you think there's a way to ship this modification as part of the extension? Would we even want to do that? Maybe good enough to add some documentation (although that's a bit clunky).

I decided to just add documentation for this. If you or I get a minute, we can research embedding that setting in this extension.

bastienboutonnet commented:
Oh DUH! Of course, I've event written those refs in yml files don't know what I was thinking haha.

Ok great correction on removing the brackets.

Regarding the comments I spent a little bit of time looking into the betterJinja repo (where I opened the ticket) and yeah I'm not sure yet. I would have to spend more time. I'm sure somethnig can be done...

In the interest of merging your work, would you mind adding a pretty visible warning in the readme maybe?

We should make a "Known Issues" section and explain about the comment shortcut. Does that sound good to you @chrisluedtke ?

chrisluedtke committed:
add known issues for comment style

chrisluedtke commented:
@bastienboutonnet done!

bastienboutonnet commented:
Awesome this looks very good! Thanks a lot for your work and patience. I'll go ahead and merge this now. I might slot in a few changes on my side and then I'll bump the version on the marketplace later today.

bastienboutonnet reviewed (approved):
Awesome this looks very good! Thanks a lot for your meticulous changes and patience. I'll go ahead and merge this now. I might slot in a few changes on my side and then I'll bump the version on the marketplace later today.
